File: 0779-k-th-symbol-in-grammar/0779-k-th-symbol-in-grammar.cpp
class Solution {
public:
    char check(char x,bool ok){
        if(ok){
            if (x=='0') return '1';
            else return '0';
        }
        return x;
    }
    char go (int n,int k){
        if (n==0) return '0';
        bool flag = (k%2==1);
        char x = go(n-1,k/2);
        return check(x,flag);
    }
    int kthGrammar(int n, int k) {
        char x = go(n-1,k-1);
        return x-'0';
    }
};

File: 0779-k-th-symbol-in-grammar/0779-k-th-symbol-in-grammar.py
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        def check(x, ok):
            if ok :
                if x==0:
                    return 1
                else :
                    return 0
            return x
        def go (n,k):
            if n==0 :
                return 0 
            ok = False
            if k%2==1:
                ok = True
            x = go(n-1,k//2)
            return check(x,ok)
        return go(n-1,k-1)